collectionfunctions.php

```diff
@@ -81,7 +93,49 @@ function checkNew(PDO $database) {
                 header('Location: collectionchallenge.php');
             }
         } else {
-            echo "<div class='error'>Unable to add Pokemon to the collection</div>";
+            echo "<div class='error'>Unable to add Pokémon to the collection</div>";
         }
     }
 }
+
+/** updates Pokémon database if values are the correct type
+ * @param PDO $database
+ */
+function editPoke(PDO $database) {
+    if (isset($_POST['name'], $_POST['type1'], $_POST['hp'], $_POST['attack'], $_POST['defense'], $_POST['spAttack'], $_POST['spDefense'], $_POST['speed'])) {
+        if (is_numeric($_POST['hp'] . $_POST['attack'] . $_POST['defense'] . $_POST['spAttack'] . $_POST['spDefense'] . $_POST['speed'])) {
+            $editPokemon = $database->prepare("UPDATE `stats` SET `name` = :name, `type1` = :type1, `type2` = :type2, `hp` = :hp, `attack` = :attack, `defense` = :defense, `spAttack` = :spAttack, `spDefense` = :spDefense, `speed` = :speed WHERE `id` = :id LIMIT 1;");
+            $ifExecute = $editPokemon->execute([
+                ':id' => $_GET['id'],
+                ':name' => $_POST['name'],
+                ':type1' => $_POST['type1'],
+                ':type2' => $_POST['type2'],
+                ':hp' => $_POST['hp'],
+                ':attack' => $_POST['attack'],
+                ':defense' => $_POST['defense'],
+                ':spAttack' => $_POST['spAttack'],
+                ':spDefense' => $_POST['spDefense'],
+                ':speed' => $_POST['speed']
+            ]);
+            if ($ifExecute) {
+                header('Location: collectionchallenge.php');
+            } else {
+                echo "<div class='error'>Unable to edit Pokémon in the collection</div>";
+            }
+        }
+    }
+}
+
+/** Soft delete Pokémon record and refresh page to show undeleted pokemon
+ * @param PDO $database
+ */
+function deletePoke(PDO $database) {
+    $deletePokemon = $database->prepare('UPDATE `stats` SET `deleted` = 1 WHERE `id` = :id LIMIT 1;');
```

Comment:
Not sure if you need the limit if you have the WHERE as your id as you should never have more than one row with the same id? :)

Reply:
I put it in as an extra check, but I can remove it :)